change/@ni-jasmine-parameterized-6f3cf07b-2ad9-4dd6-ae3c-dcf4eb9af2d9.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Notes on test failures:
Playwright 1.46 contains Chrome 128, Playwright 1.47 contains Chrome 129 (but isn't available on Nuget yet)

Tests **pass** in headless Playwright Webkit and Firefox
Tests **fail** in headless Playwright Chromium 128
Tests **pass** in headed Playwright Chromium 128, headed macOS Chrome 128 or 129 **if browser has focus**
Tests **fail** in headed Playwright Chromium 128, headed macOS Chrome 129 (didn't try 128) **if browser doesn't have focus**

We saw some new test failures in SLE when we upgraded to Playwright 1.46. In that case it was that something changed behavior in Chrome when a test opened a new tab, and the fix was that tests shouldn't open new tabs. [Bug 2823451](https://dev.azure.com/ni/DevCentral/_workitems/edit/2823451): SystemLinkShared Renovate upgrade of Playwright is blocked due to file-lib tests which open new browser tabs

Reply:
Test failures are in table tests, mostly with keyboard nav
```
FAILED Table action menu when action menu button is blurred, cell fires cell-action-menu-blur event
                       and then moving cursor slightly to right causes no column width changes
FAILED TableCell fires the cell-view-focus-in event when cell contents are focused, after the cell is already focused
FAILED TableCell fires cell-view-focus-in/cell-focus-in/cell-blur events when cell contents are focused/blurred
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, when the interactive content in the cell is focused, and then a scroll happens, the interactive content is blurred
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, when the interactive content in the cell is focused, and then a data update happens, the matching cell will be focused afterwards
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, when the interactive content in the cell is focused, pressing the given key will focus the cell: F2
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, when the interactive content in the cell is focused, pressing the given key will focus the cell: Escape
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, pressing the given key will focus the interactive cell content in the current cell: F2
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, pressing the given key will focus the interactive cell content in the current cell: Enter
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, each Shift-Tab press will focus each of the remaining tabbable elements in the row (to the left), then will move focus past the table
FAILED Table keyboard navigation with interactive columns and cell[0, 1] focused, each Tab press will focus each of the remaining tabbable elements in the row (to the right), then will move focus past the table
FAILED Table keyboard navigation with interactive columns with the first row focused, clicking interactive cell content in another row will focus that content, and an Esc key press will refocus the cell
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a cell focused, Ctrl-End focuses the cell (same column) in the last row, and Ctrl-Home focuses the cell (same column) in the 1st row
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a row focused, End focuses the last row, and Home focuses the 1st row
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a row focused, PgUp will focus a row one page (table viewport height of rows) up
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a row focused, PgDown will focus a row one page (table viewport height of rows) down
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a cell focused, if the table is scrolled away from that cell, then back to it, it is focused again
FAILED Table keyboard navigation with non-interactive columns for a large dataset (virtualized) table with cell[0, 0] focused, with a row focused, if the table is scrolled away from that row, then back to it, it is focused again
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for a collapsed group row, pressing the given key will expand the group: Enter
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for a collapsed group row, pressing the given key will expand the group: RightArrow
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for a collapsed group row, pressing the given key will expand the group: Space
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for an expanded group row, pressing the given key will collapse the group: Space
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for an expanded group row, pressing the given key will collapse the group: LeftArrow
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, for an expanded group row, pressing the given key will collapse the group: Enter
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, when multiple selection is enabled pressing RightArrow on an focused expanded group row will focus the selection checkbox
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, when multiple selection is enabled pressing LeftArrow on a focused group row selection checkbox will focus the entire group row
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, with no tabbable elements in the group row, pressing Tab will move focus past the table
FAILED Table keyboard navigation with non-interactive columns for a table with a grouped column, when a group row is focused, with no tabbable elements in the group row, pressing Shift-Tab will move focus past the table
FAILED Table keyboard navigation with non-interactive columns for a multiple-selection table with focus, when a row is focused, pressing DownArrow multiple times when the row selection checkbox is focused will focus the row selection checkbox on subsequent rows
FAILED Table keyboard navigation with non-interactive columns for a multiple-selection table with focus, when a row is focused, pressing Space will select the row
FAILED Table keyboard navigation with non-interactive columns for a multiple-selection table with focus, when a row is focused, pressing End when the row selection checkbox is focused will focus the last cell
FAILED Table keyboard navigation with non-interactive columns for a multiple-selection table with focus, when a row is focused, pressing RightArrow will focus the row selection checkbox
FAILED Table keyboard navigation with non-interactive columns for a multiple-selection table with focus, when a row is focused, pressing Home when a cell is focused will focus the row selection checkbox
FAILED Table keyboard navigation with non-interactive columns for a table with hierarchy when an expanded row is focused, pressing RightArrow will move to the 1st cell of the row (and the row stays expanded)
FAILED Table keyboard navigation with non-interactive columns for a table with hierarchy when a collapsed row is focused, pressing RightArrow will expand the row
FAILED Table keyboard navigation with non-interactive columns for a table with hierarchy when an expanded row is focused, pressing LeftArrow will collapse the row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus when a cell is focused, its action menu is visible #SkipFirefox #SkipWebkit
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus when a cell action menu is opened via mouse click, scrolling the table will close the action menu and focus the cell
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus when a cell action menu is opened via keyboard, scrolling the table will close the action menu and focus the cell
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus consecutive Enter key presses will sort the focused column (ascending, descending, no sort)
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus DownArrow key press will focus the cell in the same column in the next row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus on initial table focus, the 1st column header is focused if sortable
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus after a DownArrow key press (from 1st column header), cell[0, 0] is focused
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, when the table data changes (same or greater row count), the focused cell is maintained (row at the same index)
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, consecutive RightArrow presses focus each cell to the right in the same row, stopping at the last cell
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, with no tabbable elements in the row, pressing Shift-Tab will move focus past the table
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, from the last cell being focused, consecutive LeftArrow presses focus each cell to the left in the same row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, when the table data changes (fewer rows than previously focused row index), the focused cell resets to be for the 1st row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, pressing LeftArrow will focus the entire row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, with no tabbable elements in the row, pressing Tab will move focus past the table
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, End will focus the last cell in the same row, and Home will focus the 1st cell in the same row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, consecutive DownArrow presses focus each cell to the bottom in subsequent rows
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, clicking a cell in a new row, then a cell in the original row, will focus the clicked (original) row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus starting with cell[0, 0] focused, when the table data changes to zero rows, the focus resets to the header
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus when a row is focused, all action menus in that row are visible (and otherwise hidden) #SkipFirefox #SkipWebkit
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus when a row is focused, pressing Space has no effect when selectionMode is none (row not selected)
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus UpArrow key press will focus the cell in the same column in the previous row
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus consecutive RightArrow presses focus each header to the right, stopping at the last header
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus from the last header being focused, consecutive LeftArrow presses focus each header to the left, stopping at the 1st header
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus a row is tabindex 0 when focused, and tabindex -1 when clicked away from
FAILED Table keyboard navigation with non-interactive columns for a simple table (no hierarchy/grouping/selection) with focus Shift-Enter key presses will sort the focused column (ascending, descending, no sort) and maintain other sorted columns
FAILED Table keyboard navigation with non-interactive columns for a simple table with no columns sortable, on initial table focus, the 1st row is focused
FAILED Table keyboard navigation with non-interactive columns for a single-selection table with focus when a cell is focused, pressing Shift-Space will select the row, and pressing it again will deselect the row
FAILED Table keyboard navigation with non-interactive columns for a single-selection table with focus when a row is focused, pressing Space will select the row, and pressing it again will deselect the row
FAILED MenuButton menu interaction with menu slotted directly in menu-button should close the menu when the element loses focus
FAILED MenuButton menu interaction with menu passed through slot of additional element should close the menu when the element loses focus
FAILED TableColumnMenuButton menu button interactions shared menu menu button column can share menu with action menu
FAILED TableColumnMenuButton menu button interactions shared menu menu opens and closes when column1 menu button is clicked and then column2 menu button is clicked
FAILED TableColumnMenuButton menu button interactions scrolling the table closes the menu button
FAILED TableColumnMenuButton menu button interactions calling setData() closes the menu button
FAILED TableColumnMenuButton keyboard navigation with cell[0, 0] focused, menu button in cells are reachable via Tab
```

Reply:
Looks like at least some of these tests have always failed when running headed if the browser doesn't have focus. In that situation, calling `element.focus()` doesn't seem to have an effect. So the change with Chromium 128 seems to be that headless runs now behave like headed runs did.

Reply:
🙈

Reply:
One other note I forgot to write down but probably should since I'm backburnering this for a while:

I need to re-check that the test failures on the CI build match the list above. When I first investigated this I thought that was the case but might have been mistaken. In at least [one of the CI builds](https://github.com/ni/nimble/actions/runs/10727916154/job/29751307848#step:17:930) some of the failures were more like this:
```
[test:sequential]   Nimble radio control value accessor
[test:sequential]     when using a template-based form
[test:sequential]       ✗ sets correct initial selected value (5004ms)
[test:sequential] 	Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
[test:sequential] 	    at <Jasmine>
```

Reply:
Playwright 1.47.0 (Chromium 129.0.6668.29) came out so I tried upgrading to that. In [that build](https://github.com/ni/nimble/actions/runs/10946841304/job/30394252147) in Chrome I'm still seeing about 50 Angular test failures like the one above, plus this message:
```
[test:sequential] Chrome Headless 129.0.0.0 (Linux x86_64) ERROR
[test:sequential]   An error was thrown in afterAll
[test:sequential]   Error: Attempting to index past the total number of rendered rows
[test:sequential]       at TablePageObject.getRow (http://localhost:9878/_karma_webpack_/webpack:/nimble-components/src/table/testing/table.pageobject.ts:703:19)
[test:sequential]       at TablePageObject.getCell (http://localhost:9878/_karma_webpack_/webpack:/nimble-components/src/table/testing/table.pageobject.ts:713:26)
[test:sequential]       at TablePageObject.getRenderedCellView (http://localhost:9878/_karma_webpack_/webpack:/nimble-components/src/table/testing/table.pageobject.ts:152:27)
[test:sequential]       at TablePageObject.getRenderedCellAnchor (http://localhost:9878/_karma_webpack_/webpack:/nimble-components/src/table/testing/table.pageobject.ts:192:29)
[test:sequential]       at UserContext.<anonymous> (http://localhost:9878/_karma_webpack_/webpack:/nimble-angular/table-column/anchor/tests/nimble-table-column-anchor-navigation-guard.directive.spec.ts:77:33)
[test:sequential]       at Generator.next (<anonymous>)
[test:sequential]       at asyncGeneratorStep (http://localhost:9878/_karma_webpack_/webpack:/node_modules/@angular-devkit/build-angular/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js:3:1)
[test:sequential]       at apply (http://localhost:9878/_karma_webpack_/webpack:/node_modules/@angular-devkit/build-angular/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js:22:1)
[test:sequential]       at _ZoneDelegate.invoke (http://localhost:9878/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone.js:369:28)
[test:sequential]       at ProxyZoneSpec.onInvoke (http://localhost:9878/_karma_webpack_/webpack:/node_modules/zone.js/fesm2015/zone-testing.js:2081:39)
```

Locally I'm seeing hundreds of nimble-components tests fail in Chrome with a similar timeout. I haven't identified a pattern to those failures yet.

```
  Table delegated events
    ✗ delegates events with specified record ids (5007ms)
        Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
            at <Jasmine>
```

My suspicion is that the headless vs headed failures are a pre-existing problem that's unrelated to this upgrade.

Reply:
Looks like the playwright update is needed to update the github runner os version, see https://github.com/microsoft/playwright/issues/30368

Reply:
#2430 fixed the Angular failures. Locally I now only see a few test failures that I believe are pre-existing macOS-specific issues (mostly tests marked SkipWebkit which also fail in Chrome/FF on mac). I'll optimistically chalk the other failures descibed in this thread up to some combination of headless/headed differences, bad local state, or issues that have been resolved by other recent commits.

It couldn't hurt for another dev (@m-akinc, would you mind?) to clone this branch and ensure local test runs aren't any more unstable.

Reply:
Wanna try and do the runner upgrade in this PR as well? https://github.com/ni/nimble/pull/2420#issuecomment-2384745827 if not would be handy to have an issue / follow-up for it.

Reply:
[I tried, it failed](https://github.com/ni/nimble/actions/runs/11276146716/job/31359226268?pr=2387) 🙁
I'll revert that change and file an issue to track it.
```
[test:sequential                  ] 10 10 2024 09:39:40.167:ERROR [launcher]: ChromeHeadless stderr: [10346:10346:1010/093939.399639:FATAL:zygote_host_impl_linux.cc(128)] No usable sandbox! If you are running on Ubuntu 23.10+ or another Linux distro that has disabled unprivileged user namespaces with AppArmor, see https://chromium.googlesource.com/chromium/src/+/main/docs/security/apparmor-userns-restrictions.md. Otherwise seehttps://chromium.googlesource.com/chromium/src/+/main/docs/linux/suid_sandbox_development.md for more information on developing with the (older) SUID sandbox. If you want to live dangerously and need an immediate workaround, you can try using --no-sandbox.
[test:sequential                  ] #0 0x5555758e3cd2 base::debug::CollectStackTrace()
[test:sequential                  ] #1 0x5555758d0eae base::debug::StackTrace::StackTrace()
[test:sequential                  ] #2 0x5555758260ca logging::LogMessage::Flush()
[test:sequential                  ] #3 0x555575826c69 logging::LogMessageFatal::~LogMessageFatal()
[test:sequential                  ] #4 0x555573a3338e content::ZygoteHostImpl::Init()
[test:sequential                  ] #5 0x555574d01c1c content::ContentMainRunnerImpl::Initialize()
[test:sequential                  ] #6 0x555574cffb1c content::RunContentProcess()
[test:sequential                  ] #7 0x555574cfff47 content::ContentMain()
[test:sequential                  ] #8 0x55557094c380 ChromeMain
[test:sequential                  ] #9 0x7f11d7a2a1ca (/usr/lib/x86_64-linux-gnu/libc.so.6+0x2a1c9)
[test:sequential                  ] #10 0x7f11d7a2a28b __libc_start_main
[test:sequential                  ] #11 0x55557094c02a _start
```

Reply:
#2434

Reply:
Was about to merge but are we waiting on a Windows build and test check? @jattasNI @m-akinc ?

Reply:
I did an `npm install` and `npm run build` and that worked fine.

I ran `npm run validate` twice and the first time I had the following 1 failure:

```
[test-concurrent:nimble-components] [test-webkit ] WebKit 605.1.15 (Windows 8) ERROR
[test-concurrent:nimble-components] [test-webkit ]   An error was thrown in afterAll
[test-concurrent:nimble-components] [test-webkit ]   Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_545670/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]   @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_545670/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ] WebKit 605.1.15 (Windows 8): Executed 1138 of 5056 (skipped 6) ERROR (54.742 secs / 21.246 secs)
```

And the second time I had ten failures:

```
[test-concurrent:nimble-components] [test-webkit ]   MatrixRenderer worker
[test-concurrent:nimble-components] [test-webkit ]     × scaled Indices should be computed (32ms)
[test-concurrent:nimble-components] [test-webkit ]      ReferenceError: Can't find variable: OffscreenCanvas in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 148663)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:148663:52
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:148662:56
[test-concurrent:nimble-components] [test-webkit ]      <Jasmine>
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]   MatrixRenderer with MessageChannel needing canvas context
[test-concurrent:nimble-components] [test-webkit ]     × scaled Indices should be computed (16ms)
[test-concurrent:nimble-components] [test-webkit ]      ReferenceError: Can't find variable: OffscreenCanvas in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 161276)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:161276:52
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:161271:27
[test-concurrent:nimble-components] [test-webkit ]      <Jasmine>
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × setCanvasDimensions should set the canvas dimensions (1ms)
[test-concurrent:nimble-components] [test-webkit ]      ReferenceError: Can't find variable: OffscreenCanvas in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 161276)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:161276:52
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:161271:27
[test-concurrent:nimble-components] [test-webkit ]      <Jasmine>


[test-concurrent:nimble-components] [test-webkit ]     worker renderer action
[test-concurrent:nimble-components] [test-webkit ]       × will use WorkerRenderer after unsupported diesTable change but it will fail (16ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]   RichTextEditorMentionListbox
[test-concurrent:nimble-components] [test-webkit ]     Dynamically update mention popup items based on configuration changes
[test-concurrent:nimble-components] [test-webkit ]       - should show mention popup for multiple mention configuration elements (0ms)
[test-concurrent:nimble-components] [test-webkit ]     user mention via mention list box
[test-concurrent:nimble-components] [test-webkit ]       - setting `disabled` should close the mention popup #SkipWebkit (0ms)
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]   HoverHandler
[test-concurrent:nimble-components] [test-webkit ]     × will return the expected index when mouse moved in range from top-left (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × will return undefined when mouse moved out of range from top-right (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × will return undefined when mouse moved out of range from bottom-right (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × will return undefined when mouse moved out of range from bottom-left (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × will return the expected index when mouse moved in range from top-right (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
[test-concurrent:nimble-components] [test-webkit ]
[test-concurrent:nimble-components] [test-webkit ]     × will return the expected index when mouse moved in range from bottom-right (18ms)
[test-concurrent:nimble-components] [test-webkit ]      Unhandled promise rejection: TypeError: this.wafermap.workerCanvas.transferControlToOffscreen is not a function. (In 'this.wafermap.workerCanvas.transferControlToOffscreen()', 'this.wafermap.workerCanvas.transferControlToOffscreen' is undefined) in C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js (line 156070)
[test-concurrent:nimble-components] [test-webkit ]      @C:/Users/mraj/AppData/Local/Temp/_karma_webpack_297987/commons.js:156070:90
```

Reply:
Another weird thing is that it is creating a window that is fixed on the screen and blank:

![image](https://github.com/user-attachments/assets/48e07d88-7b79-4874-9207-8bca57bc7d3b)

Reply:
I'm going to try now

Reply:
I see essentially the same things as Milan. The WebKit tests inconsistently fail for `OffscreenCanvas` related reasons. When I run the tests (for all browsers) concurrently, I get the weird blank window (which doesn't appear in the taskbar or alt+tab window switcher, and ProcessExplorer's "Find Window Process" tool doesn't see it either). When I run `npm run test-webkit`, I don't get the blank window, and sometimes tests start failing due to timeout (in addition to the offscreen canvas errors).

Reply:
The [WebKit 18.0 release notes](https://webkit.org/blog/15865/webkit-features-in-safari-18-0/) mention fixing a number of `OffscreenCanvas` related bugs. Maybe it still has other bugs that we are running into?
![image](https://github.com/user-attachments/assets/26e874c7-135a-40d6-948d-e34b45286b08)

Reply:
I updated the PR to re-disable the wafer map tests related to #2169 that it had been enabling and updated the description to no longer resolve that issue. From a quick scan it looks like that covers most of the test failures you saw. @m-akinc @rajsite wanna try again and see if that covers all of them and also fixes the mysterious floating window?

Reply:
Did a sync, full clean, build, and `npm run validate` three times. During all the runs at some point `test-webkit` just starts timing out and stops making any progress:

```
[test-concurrent:nimble-components] [test-webkit ]     × should not navigate to hidden items (5018ms)
[test-concurrent:nimble-components] [test-webkit ]      Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL) in C:/dev/github/nimble/node_modules/karma-jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js (line 7612)
[test-concurrent:nimble-components] [test-webkit ]      <Jasmine>
```

I have to kill the WebKitWebProcess.exe, it restarts, and then tests start making progress again.

The tests start going very slow timing out one after another every 5 seconds. Also mysterious window still shows up.

Reply:
I'm still seeing a lot of weird stuff from the WebKit tests:
1. Blank window still shows up when running tests concurrently (but not when running just the WebKit tests)
2. After a while, the blank window disappears (while WebKit tests are still running), then tests start failing due to 5s timeout. The same thing seems to happen when running the WebKit tests by themselves, too, except the blank window was never visible, i.e. after a certain time (about 4 min), every test just starts timing out.
3. Tests are slooooowwww. I had improved this via a submission to Jasmine, but somehow it's slow again?

Maybe running WebKit tests on Windows is more trouble than it is worth.

Reply:
I don't think we have a clear next step planned so I'll move back to draft for now

Reply:
Think we should split out the arrow upgrade. Will do that after https://github.com/ni/nimble/pull/2474 is merged. I'll close this PR for now.